src/components/App/App.jsx

```diff
@@ -0,0 +1,83 @@
+import React from 'react'
+import TextField from '@material-ui/core/TextField'
+import usePrevious from '../../hooks/usePrevious'
+import Message from '../Message/Message'
+import { AUTHORS } from './constants'
+import './App.css'
+
+
+function App() {
+    const [messageList, setMessageList] = React.useState([])
+    const [inputValue, setInputValue] = React.useState('')
+
+    const timer = React.useRef(null)
+
+    const prevMessageList = usePrevious(messageList)
+
+    React.useEffect(() => {
+        if (
+            prevMessageList?.length < messageList.length &&
+            messageList[messageList.length - 1].author !== AUTHORS.BOT
+        ) {
+            timer.current = setTimeout(
```

Comment:
https://stackoverflow.com/questions/53090432/react-hooks-right-way-to-clear-timeouts-and-intervals